addons/udes_stock/models/stock_move.py

```diff
@@ -468,9 +468,9 @@ def _prepare_extra_info_for_new_picking_for_group(self, pickings, moves):
         if len(partners) == 1:
             values["partner_id"] = partners.id
 
-        dates_done = list(set(moves.mapped("date")))
-        if len(dates_done) == 1:
-            values["date_done"] = dates_done[0]
+        dates_done = set(moves.mapped("date"))
+        if set(moves.mapped("state")).issubset({"done", "cancel"}):
```

Comment:
Q: Why do we check for cancelled moves here?

Reply:
It is because if a picking has 2 moves, they cancel one then do the second - then it should have the date_done copied across. But i have no idea if somehow you can copy all cancelled moves into a new picking - i hope that doesn't happen in Odoo :)

Reply:
Ah okay, so if that means if we have a picking with 1 move cancelled and the other done, then `issubset({"done"})` would filter it out, hence why we need to allow cancel too?